proxy-node-gateway/src/test/java/uk/gov/ida/notification/apprule/base/GatewayAppRuleTestBase.java

```diff
@@ -1,22 +1,103 @@
 package uk.gov.ida.notification.apprule.base;
 
 import io.dropwizard.testing.ConfigOverride;
-
+import org.glassfish.jersey.internal.util.Base64;
 import org.junit.ClassRule;
 import org.junit.Rule;
-
+import org.opensaml.core.config.InitializationService;
+import org.opensaml.saml.saml2.core.AuthnRequest;
+import org.opensaml.security.credential.Credential;
+import uk.gov.ida.notification.VerifySamlInitializer;
+import uk.gov.ida.notification.apprule.rules.EidasSamlParserClientRule;
 import uk.gov.ida.notification.apprule.rules.GatewayAppRule;
 import uk.gov.ida.notification.apprule.rules.TranslatorClientRule;
+import uk.gov.ida.notification.apprule.rules.VerifyServiceProviderClientRule;
+import uk.gov.ida.notification.saml.SamlFormMessageType;
+import uk.gov.ida.notification.saml.SamlObjectMarshaller;
+import uk.gov.ida.notification.saml.SamlObjectSigner;
+import uk.gov.ida.saml.core.test.TestCredentialFactory;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Form;
+import javax.ws.rs.core.NewCookie;
+import javax.ws.rs.core.Response;
+import java.net.URISyntaxException;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.fail;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.TEST_RP_PRIVATE_SIGNING_KEY;
+import static uk.gov.ida.saml.core.test.TestCertificateStrings.TEST_RP_PUBLIC_SIGNING_CERT;
 
 public class GatewayAppRuleTestBase {
 
+    private Map<String, NewCookie> cookies;
+
+    protected static final String CONNECTOR_NODE_ENTITY_ID = "http://connector-node:8080/ConnectorResponderMetadata";
+
+    @ClassRule
+    public static final TranslatorClientRule translatorClientRule;
+
+    @ClassRule
+    public static final EidasSamlParserClientRule espClientRule = new EidasSamlParserClientRule();
+
     @ClassRule
-    public static final TranslatorClientRule translatorClientRule = new TranslatorClientRule();
+    public static final VerifyServiceProviderClientRule vspClientRule = new VerifyServiceProviderClientRule();
+
+    static {
+        try {
+            InitializationService.initialize();
+            VerifySamlInitializer.init();
+            translatorClientRule = new TranslatorClientRule();
```

Comment:
Can this just be inlined on line 38 like the other client rules? Also, do we need the try, catch? If they throw an exception the test will bork anyway?

Reply:
Yes, inlined